runtime/gc_vlhgc/GlobalMarkingScheme.cpp

```diff
@@ -1128,6 +1128,35 @@ MM_GlobalMarkingScheme::scanContinuationObjects(MM_EnvironmentVLHGC *env)
 	env->getGCEnvironment()->_continuationObjectBuffer->flush(env);
 }
 
+void
+MM_GlobalMarkingScheme::iterateContinuationObjects(MM_EnvironmentVLHGC *env)
+{
+	J9JavaVM *vm = (J9JavaVM*)env->getOmrVM()->_language_vm;
+	J9JITConfig *jitConfig = vm->jitConfig;
+	if (NULL == jitConfig->methodsToDelete) {
+		return;
+	}
+
+	MM_HeapRegionDescriptorVLHGC *region = NULL;
+	GC_HeapRegionIteratorVLHGC regionIterator(_heapRegionManager);
+	while (NULL != (region = regionIterator.nextRegion())) {
+		if (region->containsObjects()) {
+			if (NULL != region->getContinuationObjectList()->getHeadOfList()) {
+				if (J9MODRON_HANDLE_NEXT_WORK_UNIT(env)) {
+					J9Object *object = region->getContinuationObjectList()->getHeadOfList();
+					while (NULL != object) {
+						Assert_MM_true(region->isAddressInRegion(object));
+						J9Object* next = _extensions->accessBarrier->getContinuationLink(object);
+						TRIGGER_J9HOOK_MM_WALKCONTINUATION(_extensions->hookInterface, (J9VMThread *)env->getLanguageVMThread(), object);
+						object = next;
+					}
+				}
+			}
+		}
+	}
+	_extensions->didIterateContinuationListForJIT = true;
+}
```

Comment:
actually, a cleaner approach is to put it into HeapRegionManager (it's higher level abstract than Buffer), since iteration assumes some knowledge about how regions are organized, and indeed we do have separate HRMs for Standard and VLHGC configurations

Reply:
both MM_HeapRegionManager and MM_HeapRegionManagerStandard are in omr, if we put it into HeapRegionManager, we need to update omr code

Reply:
just realized it. nah, scratch it. we should not be introducing Java spacific code to OMR

Reply:
ok,  I will remove debug code